src/rospack.cpp

```diff
@@ -2112,20 +2112,30 @@ Rosstackage::validateCache()
     cache_max_age = atof(user_cache_time_str);
   if(cache_max_age == 0.0)
     return NULL;
-  struct stat s;
-  if(stat(cache_path.c_str(), &s) == 0)
-  {
-    double dt = difftime(time(NULL), s.st_mtime);
-    // Negative cache_max_age means it's always new enough.  It's dangerous
-    // for the user to set this, but rosbash uses it.
-    if ((cache_max_age > 0.0) && (dt > cache_max_age))
-      return NULL;
-  }
+  struct stat ls;
+  if(lstat(cache_path.c_str(), &ls) == -1)
+    return NULL;
+
+  double dt = difftime(time(NULL), ls.st_mtime);
+  // Negative cache_max_age means it's always new enough.  It's dangerous
+  // for the user to set this, but rosbash uses it.
+  if ((cache_max_age > 0.0) && (dt > cache_max_age))
+    return NULL;
+
   // try to open it
   FILE* cache = fopen(cache_path.c_str(), "r");
   if(!cache)
     return NULL; // it's not readable by us. sad.
 
+  struct stat s;
+  if(fstat(fileno(cache), &s) == -1)
+    return NULL;
```

Comment:
Doesn't this need a `fclose(cache)` here?

Reply:
Yes you are right. Sorry missed this.

Reply:
Thank you very much for helping fix this :)